nixos/modules/services/networking/xray.nix

```diff
@@ -80,9 +80,13 @@ with lib;
       description = "xray Daemon";
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
+      script = ''
+        cd $CREDENTIALS_DIRECTORY
+        exec "${cfg.package}/bin/xray" -config config.json
```

Comment:
`exec "${cfg.package}/bin/xray" -config "$CREDENTIALS_DIRECTORY/config.json"`
avoids changing the working directory